### PR TITLE
Gather performance baselines for 5.10 host/guest kernel version on all platforms

### PR DESCRIPTION
**Pull request by dianpopa**
# Reason for This PR

Obtain performance baseline on supported host/guest kernel versions.

## Description of Changes

* added a built-in tool for gathering baselines; the tool recreates baselines for an existent performance
    test based on the old baseline
*  the name of the baselines file now contains the host kernel
    on which they were obtained
* added per cpu baselines for snapshot performance test
* run all long running performance tests on all guest supported guest kernels
* the dumped results now also contain the host kernel version
* 
- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [ ] The issue which led to this PR has a clear conclusion.
- [ ] This PR follows the solution outlined in the related issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes follow the [Runbook for Firecracker API changes][2].
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -129,28 +131,88 @@ def _test_images_s3_bucket():
...
+
+    @staticmethod
+    def __dump_pretty_json(file, data, flags):
+        """Write the `data` dictionary to the output file in pretty format."""
+        with open(file, flags) as file_fd:
+            json.dump(data, file_fd, indent=4)
+            file_fd.write("\n")  # Add newline cause Py JSON does not
+
+    # pylint: disable=R1702
+    def __gen_baselines(self, result_dict):
```
Comment (gc-plp) [inline tests/conftest.py:172]: Is there any reason for using double underscore for methods? This instructs the python compiler to rewrite the attribute name to `_ResultsFileDumper__gen_baselines_`. 

For reference, see https://www.python.org/dev/peps/pep-0008 `__double_leading_underscore: when naming a class attribute, invokes name mangling (inside class FooBar, __boo becomes _FooBar__boo`

File: tests/conftest.py [hunk 2]
```diff
@@ -129,28 +131,88 @@ def _test_images_s3_bucket():
...
+                        result_node = result_dict["results"][result]
+                        function_node = result_node[measure][fun]
+                        value = round(function_node["value"])
+
+                        kernel_node = baselines[measure][kernel]
+                        vm_cfg_node = kernel_node[dist][vm_cfg]
+
+                        # Some tests do not have a workload node (i.e snapshot
+                        # performance.
+                        if fun in vm_cfg_node:
```
Comment (gc-plp) [inline tests/conftest.py:218]: Adding 
```py
if fun not in vm_cfg_node:
    continue
```

Should get rid of the `pylint: disable=R1702` exception at the top.

File: tests/conftest.py [hunk 3]
```diff
@@ -244,8 +311,16 @@ def test_session_tmp_path(test_fc_session_root_path):
 @pytest.fixture
 def results_file_dumper(request):
     """Yield the custom --dump-results-to-file test flag."""
+    generate_baselines = False
+    dump_results = False
+
     if request.config.getoption("--dump-results-to-file"):
-        return ResultsFileDumper(request.node.originalname)
+        dump_results = True
+    if request.config.getoption("--generate-baselines"):
```
Comment (gc-plp) [inline tests/conftest.py:334]: `getoption` has a `default` parameter that you can use:

`generate_baselines = request.config.getoption("--generate-baselines", default=False)`

File: tests/integration_tests/performance/test_vsock_throughput.py [hunk 4]
```diff
@@ -350,7 +357,10 @@ def iperf_workload(context):
     except core.CoreException as err:
         handle_failure(file_dumper, err)
 
-    dump_test_result(file_dumper, result)
+    # This will dump the things we told them to dump
+    # in the pytest provided parameters.
+    if file_dumper:
```
Comment (gc-plp) [inline tests/integration_tests/performance/test_vsock_throughput.py:361]: Proposal: as noted in a previous comment, the `file_dumper` parameter could be an instance that implements an early exit for the `dump()` call.

This way you can avoid checking for `if file_dumper`.

File: tests/conftest.py [hunk 5]
```diff
@@ -244,8 +311,16 @@ def test_session_tmp_path(test_fc_session_root_path):
...
+        generate_baselines = True
+
+    if dump_results or generate_baselines:
+        return ResultsFileDumper(request, dump_results, generate_baselines)
```
Comment (gc-plp) [inline tests/conftest.py:338]: This method could always return a `ResultsFileDumper` instance with a dummy functionality for `dump` whenever dumping is not enabled.

File: tests/integration_tests/performance/test_snapshot_restore_performance.py [hunk 6]
```diff
@@ -62,8 +62,15 @@ class SnapRestoreBaselinesProvider(BaselineProvider):
 
     def __init__(self, env_id):
         """Snapshot baseline provider initialization."""
-        baselines = CONFIG["hosts"]["instances"]["m5d.metal"]
-        super().__init__(DictQuery(baselines))
+        cpu_model_name = get_cpu_model_name()
+        print("cpu model name {}".format(cpu_model_name))
```
Comment (gc-plp) [inline tests/integration_tests/performance/test_snapshot_restore_performance.py:70]: Extra print statement.

File: tests/conftest.py [hunk 7]
```diff
@@ -202,6 +202,14 @@ def __gen_baselines(self, result_dict):
                         else:
                             value = round(function_node["value"])
 
+                        # It may happen that the results contain a kernel
+                        # version for which the config does not have an entry.
+                        # It could be that the S3 bucket contains an outdated
+                        # or new kernel version.
+                        assert kernel in baselines[measure],\
+                            "Results contain kernel version for which the" \
```
Comment (alindima) [inline tests/conftest.py:210]: let's also output the said kernel version

File: tests/integration_tests/performance/configs/test_block_performance_config_5.10.json [hunk 8]
```diff
@@ -275,39 +275,39 @@
                                         "1vcpu_1024mb.json": {
                                             "Avg": {
                                                 "randrw-bs4096": {
-                                                    "target": 83319,
+                                                    "target": 58061,
```
Comment (alindima) [inline tests/integration_tests/performance/configs/test_block_performance_config_5.10.json:438]: hm it's weird that the performance numbers reported here are worse with io_uring.
I'm fairly sure that this is due to the fact that we run those tests using a ramdisk.

We should gather those baselines on a nvme drive for io_uring.
If this is too much work for this PR, we may hardcode the use of the Sync backend in this test and work on it later

File: tests/framework/stats/core.py [hunk 9]
```diff
@@ -61,9 +62,13 @@ def add_pipe(self, producer: Producer, consumer: Consumer, tag=None):
     def run_exercise(self, fail_fast=False) -> Result:
         """Drive the statistics producers until completion."""
         iterations = self._result['iterations']
+        cpu_model = self._result['custom']['cpu_model_name']
         # This is used for indentation purposes.
 
         for tag, pipe in self._pipes.items():
+            self._logger.info("Running workload {} on processor {}".format(
```
Comment (alindima) [inline tests/framework/stats/core.py:69]: this message still won't be visible since we have `log_cli_level = ERROR` in `pytest.ini`.
I don't know what the best solution would be here. What about having this as a `warn` and setting the `log_cli_level=WARN`

File: tests/framework/artifacts.py [hunk 10]
```diff
@@ -304,7 +304,7 @@ class ArtifactCollection:
     PLATFORM = platform.machine()
 
     # S3 bucket structure.
-    ARTIFACTS_ROOT = 'ci-artifacts'
+    ARTIFACTS_ROOT = 'ci-artifacts-dev'
```
Comment (alindima) [inline tests/framework/artifacts.py:307]: let's maybe also add the temp docker container tag in this commit, to make sure that we don't forget abt it

File: tests/integration_tests/performance/configs/test_vsock_throughput_config_4.14.json [hunk 11]
```diff
@@ -0,0 +1,962 @@
+{
```
[the remark this answers is not in the document]
Comment (alindima) [reply tests/integration_tests/performance/configs/test_vsock_throughput_config_4.14.json:1]: should we have a commit removing them?

File: tests/conftest.py (hunk 5 above)
Comment (gc-plp) [reply tests/conftest.py:338]: @alindima actually, I've given this a second thought and it's not really needed to add the checks anywhere in the tests.

Hooks can be added when tests start/end, like in the JSON test report thing, and the dumping can be done implicitly.

File: tests/framework/stats/core.py (hunk 9 above)
Comment (gc-plp) [reply tests/framework/stats/core.py:69]: I think that you can override the log level for a specific logger instance.

File: tests/integration_tests/performance/configs/test_block_performance_config_5.10.json (hunk 8 above)
Comment (gc-plp) [reply tests/integration_tests/performance/configs/test_block_performance_config_5.10.json]: NVMe breaks other things unfortunatelly.

Ideally there should be different profiles for each environment (i.e. ramdisk, nvme, etc.).

File: tests/integration_tests/performance/test_vsock_throughput.py (hunk 4 above)
Comment (dianpopa) [reply tests/integration_tests/performance/test_vsock_throughput.py]: Done.

File: tests/integration_tests/performance/test_snapshot_restore_performance.py (hunk 6 above)
Comment (dianpopa) [reply tests/integration_tests/performance/test_snapshot_restore_performance.py]: Removed it.

File: tests/integration_tests/performance/test_snapshot_restore_performance.py [hunk 12]
```diff
@@ -62,8 +62,15 @@ class SnapRestoreBaselinesProvider(BaselineProvider):
...
+        baselines = list(filter(
+            lambda cpu_baseline: cpu_baseline["model"] == cpu_model_name,
+            CONFIG["hosts"]["instances"]["m5d.metal"]["cpus"]))
+
+        super().__init__(DictQuery(dict()))
+        if len(baselines) > 0:
+            super().__init__(DictQuery(baselines[0]))
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/test_snapshot_restore_performance.py:76]: Indeed, but It s everywhere like this.

File: tests/integration_tests/performance/test_network_latency.py [hunk 13]
```diff
@@ -230,4 +230,7 @@ def _g2h_send_ping(context):
     except core.CoreException as err:
         handle_failure(file_dumper, err)
 
-    dump_test_result(file_dumper, result)
+        # This will dump the things we told them to dump
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/test_network_latency.py]: ok, take a look.

File: tests/integration_tests/performance/test_snapshot_perf.py [hunk 14]
```diff
@@ -565,7 +565,7 @@ def test_snapshot_resume_latency(network_config,
     microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="2vcpu_512mb"))
     microvm_artifacts.insert(artifacts.microvms(keyword="2vcpu_256mb"))
 
-    kernel_artifacts = ArtifactSet(artifacts.kernels(keyword="4.14"))
+    kernel_artifacts = ArtifactSet(artifacts.kernels())
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/test_snapshot_perf.py:565]: I do not quite understand the first phrase. I took care of making sure to test on all kernels the performance tests. I was targeting to only touch down on perf tests, indeed.

File: tests/conftest.py (hunk 7 above)
Comment (dianpopa) [reply tests/conftest.py]: No longer applies.

File: tests/framework/stats/core.py (hunk 9 above)
Comment (dianpopa) [reply tests/framework/stats/core.py:68]: We can set --log-cli-level=INFO in the pipelines which is what I did.

File: tests/framework/utils_cpuid.py [hunk 15]
```diff
@@ -26,10 +30,26 @@ def get_cpu_vendor():
...
+    if platform.machine() == "aarch64":
+        _, stdout, _ = run_cmd("cat /proc/cpuinfo | grep 'CPU part' | uniq")
+    else:
+        _, stdout, _ = run_cmd("cat /proc/cpuinfo | grep 'model name' | uniq")
     info = stdout.strip().split(sep=":")
     assert len(info) == 2
-    return info[1].strip()
+    raw_cpu_model = info[1].strip()
+    if "Intel" in raw_cpu_model:
+        return raw_cpu_model
```
Comment (serban300) [inline tests/framework/utils_cpuid.py:41]: Does this only apply for `Intel` or for `AMD` as well ?

File: tools/devtool [hunk 16]
```diff
@@ -72,7 +72,7 @@
 DEVCTR_IMAGE_NO_TAG="public.ecr.aws/firecracker/fcuvm"
 
 # Development container tag
-DEVCTR_IMAGE_TAG="v33"
+DEVCTR_IMAGE_TAG="kernel_support"
```
Comment (serban300) [inline tools/devtool:75]: Is this intentional ?

File: tests/framework/artifacts.py [hunk 17]
```diff
@@ -420,14 +420,22 @@ def firecracker_versions(self, min_version=None, max_version=None):
...
             keyword=keyword
         )
 
+        valid_kernels = []
+        for s3_kernel in kernels:
+            for sup_kernel in SUPPORTED_KERNELS:
+                if sup_kernel in s3_kernel.key:
+                    valid_kernels.append(s3_kernel)
+                    continue
+        return valid_kernels
```
[the remark this answers is not in the document]
Comment (serban300) [reply tests/framework/artifacts.py:437]: Looks good, but instead of splitting isn't there a `SUPPORTED_KERNELS.any(|key| kernel.key in key)` ?

File: tests/conftest.py [hunk 18]
```diff
@@ -132,25 +133,31 @@ def _test_images_s3_bucket():
 class ResultsFileDumper:  # pylint: disable=too-few-public-methods
     """Class responsible with outputting test results to files."""
 
-    def __init__(self, test_name: str, append=True):
+    def __init__(self, request, dump_results):
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/conftest.py:136]: Well this class used also contain the generation of the baselines. 

Would you be ok to switching to the old one? @alindima 

My intention was to improve this, but actually brought along more effort to this PR.

File: tests/framework/utils.py [hunk 19]
```diff
@@ -611,14 +611,16 @@ def compare_versions(first, second):
...
-def get_kernel_version():
+def get_kernel_version(include_patch=True):
     """Return the current kernel version in format `major.minor.patch`."""
     linux_version = platform.release()
     for idx, char in enumerate(linux_version):
         if not char.isdigit() and char != '.':
             linux_version = linux_version[0:idx]
+            if not include_patch:
+                major, minor, *_rest = linux_version.split(".")
+                linux_version = f"{major}.{minor}"
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/framework/utils.py:622]: I understand you suggest to have default level 2. That means I need to change all the code that used it before this PR which used level 3. This would make this PR even more cumbersome.

File: tests/framework/artifacts.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/framework/artifacts.py:437]: I do not see how you make use of the kernel files from S3 in your filtering proposal. I need a more verbose replacing expression.

File: tools/devtool (hunk 16 above)
Comment (dianpopa) [reply tools/devtool:75]: yes, I used a development image for this PR.

In my next push, I am replacing it with v34 (the next one after the current one).

File: tests/integration_tests/performance/test_block_performance.py [hunk 20]
```diff
@@ -366,8 +366,11 @@ def fio_workload(context):
     try:
         result = st_core.run_exercise()
     except core.CoreException as err:
+        basevm.kill()
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/test_block_performance.py]: No, the destructor is not called and the disks could potentially fill up. But this requires more in depth work.

File: tests/framework/artifacts.py (hunk 10 above)
Comment (dianpopa) [reply tests/framework/artifacts.py]: I switched to using the production bucket.

File: tests/framework/builder.py [hunk 21]
```diff
@@ -142,7 +142,8 @@ def build(self,
             drive_id='rootfs',
             path_on_host=jailed_rootfs_path,
             is_root_device=True,
-            is_read_only=False
+            is_read_only=False,
+            io_engine=microvm_config.get("io_engine", None)
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/framework/builder.py]: I changed this to something similar to what you suggested. Take a look.

File: tests/integration_tests/performance/configs/test_block_performance_config_5.10.json [hunk 22]
```diff
@@ -131,6 +131,46 @@
                                                     "delta_percentage": 7
                                                 }
                                             }
+                                        },
+                                        "syncio_1vcpu_1024mb.json": {
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/configs/test_block_performance_config_5.10.json]: If we use separate files, a looot of the logic found in block_performance would be duplicated.

Changing the structure of the dict is tricky since the comparison criteria is based on the structure of the config file.

File: tests/framework/utils.py (hunk 19 above)
[the remark this answers is not in the document]
Comment (serban300) [reply tests/framework/utils.py:622]: I thought level 2 == patch. Because The levels start from 0.

File: tests/conftest.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (serban300) [reply tests/conftest.py:136]: Yes, I would change it. But I think you could do the change in a separate commit at the end in order to avoid modifying all the commits.

[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/conftest.py:136]: What is the proposal exactly? ResultsFileDumper is obtained by calling into a fixture and each perf test has an instance of it. How would it look for the NopDumper? Would it have a dump method? Does not this option requrie additional if/else checks inside the tests?

**Review by dianpopa — COMMENTED** [review]
@alindima @serban300 please take another look! I might have missed some comments but please leave them unresolved and I will get back.

File: tests/framework/utils_cpuid.py (hunk 15 above)
Comment (serban300) [reply tests/framework/utils_cpuid.py]: Can we simply test if the platform is `x86` ? Instead of testing for Intel and AMD.

File: tests/integration_tests/performance/test_snapshot_restore_performance.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (alindima) [reply tests/integration_tests/performance/test_snapshot_restore_performance.py]: yeah, I'm fine either way

File: tests/framework/stats/core.py (hunk 9 above)
Comment (alindima) [reply tests/framework/stats/core.py]: I see, this is fine then

File: tests/conftest.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (serban300) [reply tests/conftest.py:136]: The proposal is something like:

```
class ResultsDumperInterface:
    def dump(self, result):
        pass

class NopResultsDumper(ResultsDumperInterface):
    def dump(self, result):
        pass

class JsonFileDumper(ResultsDumperInterface):  # pylint: disable=too-few-public-methods
    """Class responsible with outputting test results to files."""

    def __init__(self, request):
        """Initialize the instance."""
        self._results_file = None

        test_name = request.node.originalname
        self._root_path = defs.TEST_RESULTS_DIR
        # Create the root directory, if it doesn't exist.
        self._root_path.mkdir(exist_ok=True)
        self._results_file = os.path.join(
            self._root_path, "{}_results_{}.json".format(
                test_name, utils.get_kernel_version()))

    @staticmethod
    def __dump_pretty_json(file, data, flags):
        """Write the `data` dictionary to the output file in pretty format."""
        with open(file, flags, encoding='utf-8') as file_fd:
            json.dump(data, file_fd, indent=4)
            file_fd.write("\n")  # Add newline cause Py JSON does not
            file_fd.flush()

    def dump(self, result):
        """Dump the results in JSON format."""
        self.__dump_pretty_json(self._results_file, result, "a")

@pytest.fixture
def results_file_dumper(request):
    """Yield the custom --dump-results-to-file test flag."""
    if request.config.getoption("--dump-results-to-file"):
        return JsonFileDumper(request)

    return NopResultsDumper(request)
```

I didn't test it, but something along these lines.

File: tests/framework/utils.py (hunk 19 above)
[the remark this answers is not in the document]
Comment (serban300) [reply tests/framework/utils.py]: Sorry, actually to deduplicate a bit I think we can do something like:

```
        if char == '.':
            actual_level += 1
        if actual_level > level or (not char.isdigit() and char != '.')
            linux_version = linux_version[0:idx]
            break
```

File: tests/integration_tests/performance/configs/test_vsock_throughput_config_4.14.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/configs/test_vsock_throughput_config_4.14.json:1]: > the 4.14 files should just be renames of the old config files right?

not quite since we also added support for guest kernel 5.10 and we also added support for running on ARM.

File: tests/framework/stats/core.py [hunk 23]
```diff
@@ -61,9 +62,13 @@ def add_pipe(self, producer: Producer, consumer: Consumer, tag=None):
     def run_exercise(self, fail_fast=False) -> Result:
         """Drive the statistics producers until completion."""
         iterations = self._result['iterations']
+        cpu_model = self._result['custom']['cpu_model_name']
```
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/framework/stats/core.py]: I removed it and logged the processor information from inside the tests.

File: tests/conftest.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/conftest.py]: Implemented the changes.

**Review by dianpopa — COMMENTED** [review]
Addressed latest comments

File: tests/conftest.py [hunk 24]
```diff
@@ -129,28 +130,50 @@ def _test_images_s3_bucket():
...
+    def dump(self, result):
+        """Dump the results in JSON format."""
+
+
+# pylint: disable=too-few-public-methods
+class NopResultsDumper(ResultsDumperInterface):
+    """Interface for dummy dumping results to file."""
+
+    def dump(self, result):
+        """Dump the results in JSON format."""
```
Comment (serban300) [inline tests/conftest.py:146]: Nit: Outdated comment. This method doesn't do anything.

**Review by serban300 — APPROVED** [review]

File: tests/integration_tests/performance/test_snapshot_perf.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (alindima) [reply tests/integration_tests/performance/test_snapshot_perf.py:565]: I mean that there are multiple performance tests in this file that only run with guest 4.14.
Like: test_snapshot_create_diff_latency, test_snapshot_create_old_latency, test_older_snapshot_resume_latency

File: tests/framework/utils_cpuid.py [hunk 25]
```diff
@@ -26,10 +30,26 @@ def get_cpu_vendor():
...
+    return ARM_CPU_DICT[raw_cpu_model]
+
+
+def get_instance_type():
+    """Get the instance type based on the processor."""
+    proc_model = proc.proc_type()
+    instance_type = {"Intel": "m5d.metal", "ARM": "m6g.metal"}
```
[the remark this answers is not in the document]
Comment (alindima) [reply tests/framework/utils_cpuid.py:48]: you think it's a good idea using IMDS instead?

File: tests/integration_tests/performance/test_network_latency.py [hunk 26]
```diff
@@ -85,7 +103,7 @@ def consume_ping_output(cons, raw_data, requests):
 
     # E.g: round-trip min/avg/max/stddev = 17.478/17.705/17.808/0.210 ms
     stat_values = output[-1]
-    pattern_stats = "round-trip min/avg/max/stddev = (.+)/(.+)/(.+)/(.+) ms"
+    pattern_stats = "min/avg/max/[a-z]+dev = (.+)/(.+)/(.+)/(.+) ms"
```
Comment (alindima) [inline tests/integration_tests/performance/test_network_latency.py:106]: why was this change needed?

File: tests/integration_tests/performance/configs/test_vsock_throughput_config_4.14.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (alindima) [reply tests/integration_tests/performance/configs/test_vsock_throughput_config_4.14.json:1]: makes sense, but let's remove the configs that are no longer used (the ones that don't include the host kernel version in their names)

File: tests/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alindima) [reply tests/conftest.py]: Hm, but I see no modifications to the baseline parsers. I believe they need some updates to keep them running with the latest changes (given the new test names and possibly the new dimension added by the host kernel version).
Do you also think it makes sense to add said flag to them that only updates the existing baselines instead of regenerating them altogether?

File: tests/integration_tests/performance/test_snapshot_perf.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (dianpopa) [reply tests/integration_tests/performance/test_snapshot_perf.py:565]: Indeed I missed those, I only did it for long running performance tests.

I added a new commit where I also test short running CI with all supported guest kernels.

**Comment by dianpopa** [comment]
@serban300, @alindima I additionally added a new commit where I delete old config files and another one where the short running CI tests are run against all supported guest kernels. PTAL!

**Comment by alindima** [comment]
Looks good to me.
I have one more comment:

> Hm, but I see no modifications to the baseline parsers. I believe they need some updates to keep them running with the latest changes (given the new test names and possibly the new dimension added by the host kernel version).
Do you also think it makes sense to add said flag to them that only updates the existing baselines instead of regenerating them altogether?

**Review by serban300 — APPROVED** [review]

**Comment by dianpopa** [comment]
> Looks good to me. I have one more comment:
> 
> > Hm, but I see no modifications to the baseline parsers. I believe they need some updates to keep them running with the latest changes (given the new test names and possibly the new dimension added by the host kernel version).
> > Do you also think it makes sense to add said flag to them that only updates the existing baselines instead of regenerating them altogether?

@alindima would it be ok to open an issue and do this in another PR? I d firstly need to get acquainted with the parse_baselines tool and this will further block this PR.

**Comment by alindima** [comment]
> > Looks good to me. I have one more comment:
> > > Hm, but I see no modifications to the baseline parsers. I believe they need some updates to keep them running with the latest changes (given the new test names and possibly the new dimension added by the host kernel version).
> > > Do you also think it makes sense to add said flag to them that only updates the existing baselines instead of regenerating them altogether?
> 
> @alindima would it be ok to open an issue and do this in another PR? I d firstly need to get acquainted with the parse_baselines tool and this will further block this PR.

Yes, opening a new issue is fine. I'd like us to make sure that we'll work on it in the near future though

**Review by alindima — DISMISSED** [review]

**Review by serban300 — APPROVED** [review]

**Review by alindima — APPROVED** [review]
